torch/_inductor/codegen/cpp_flex_attention_template.py

```diff
@@ -0,0 +1,1093 @@
+                  j * eheadSize * kvSize + n * eheadSize),
+          /* ld_dst */ cur_kvSplitSize);
+      }
+    // Move to the next query
+    at::native::data_index_step(i, batchSize, j, num_head, l, kvSlice);
+    }
+  });
+
+  // Attention loop below
+  at::parallel_for(0, batchSize * num_head * qSlice, 1, [&](int64_t begin, int64_t end) {
+    int64_t i = 0, j = 0, k = 0;
+    at::native::data_index_init(begin, i, batchSize, j, num_head, k, qSlice);
+    int ompIdx = at::get_thread_num();
+    accum_t* buf_ptr = buf_data + ompIdx * _size_per_thread;
+    accum_t* qk_data = buf_ptr;
+    accum_t* qk_max_data = qk_data + qSplitSize * kvSplitSize;
+    accum_t* qk_sum_data = qk_max_data + qSplitSize;
+    accum_t* dst_data = qk_sum_data + qSplitSize;
+    scalar_t *qk_reduced_data =
+        is_reduced_type
+            ? buf_reduced_data + ompIdx * qSplitSize * ekvSplitSize
+            : nullptr;
+    scalar_t* query_t_padding_ptr = (!headSize_even && need_pack)
+            ? query_padding_ptr + ompIdx * qSplitSize * eheadSize
+            : nullptr;
+
+    for ([[maybe_unused]] auto z : c10::irange(begin, end)) {
+      int64_t m = k * qSplitSize;
+      int64_t cur_qSplitSize = std::min(qSplitSize, qSize - m);
+      // Initialize max and sum
+      {{kernel.kernel_name}}_fill_stub(qk_max_data,
+          -std::numeric_limits<accum_t>::infinity(), cur_qSplitSize);
+      {{kernel.kernel_name}}_fill_stub(qk_sum_data,
+          static_cast<accum_t>(0), cur_qSplitSize);
+
+      if (!headSize_even && need_pack) {
+        // Pad query if headSize is not even
+        {{kernel.kernel_name}}_copy_value_with_pad<scalar_t>(
+          q_data + i * qStrideB + j * qStrideH + m * qStrideM,
+          query_t_padding_ptr,
+          cur_qSplitSize,
+          headSize,
+          cur_qSplitSize,
+          eheadSize,
+          qStrideM
+        );
+      }
+      for (int64_t n = 0; n < kvSize; n += kvSplitSize) {
+        int64_t cur_kvSplitSize = std::min(kvSplitSize, kvSize - n);
+        int64_t cur_ekvSplitSize = (need_pack && cur_kvSplitSize % 2 != 0) ? cur_kvSplitSize + 1 : cur_kvSplitSize;
+
+        // Calculate scale * q @ k.T
+        auto i_kv = is_broadcast_bs_kv ? i/bs_shards : i;
+        auto j_kv = is_broadcast_head_kv ? j/gqa_shards : j;
+        auto kv_block_num = n / kvBlockSize;
+        auto kv_block_offset = n - kv_block_num * kvBlockSize;
+        // getting kv indices by [BS, Head, 1, kv_block_num]
+        auto i_kvi = is_broadcast_bs_kvi ? i/bs_shards_kvi : i;
+        auto j_kvi = is_broadcast_head_kvi ? j/gqa_shards_kvi : j;
+        auto kv_logical_data = kv_indices_data + i_kvi * kviStrideB +
+                                j_kvi * kviStrideH + kv_block_num;
+        if (!need_pack) {
+          auto k_addr_t = key_reorder_ptr + i * num_head * eheadSize * kvSize +
+                  j * eheadSize * kvSize + n * eheadSize;
+          at::native::cpublas::brgemm(
```

Comment:
Add a TODO here - we should use the micro-gemm template rather than invoking the brgemm interfaces directly in the future.

Reply:
Added TODO here, thanks.